zcash_proofs/src/lib.rs

```diff
@@ -169,28 +375,48 @@ pub fn parse_parameters<R: io::Read>(
     // want to read it, though, so that the BLAKE2b computed afterward is consistent
     // with `b2sum` on the files.
     let mut sink = io::sink();
-    io::copy(&mut spend_fs, &mut sink)
-        .expect("couldn't finish reading Sapling spend parameter file");
-    io::copy(&mut output_fs, &mut sink)
-        .expect("couldn't finish reading Sapling output parameter file");
-    if let Some(mut sprout_fs) = sprout_fs.as_mut() {
-        io::copy(&mut sprout_fs, &mut sink)
-            .expect("couldn't finish reading Sprout groth16 parameter file");
-    }
 
-    if spend_fs.into_hash() != SAPLING_SPEND_HASH {
-        panic!("Sapling spend parameter file is not correct, please clean your `~/.zcash-params/` and re-run `fetch-params`.");
-    }
+    // TODO: use the correct paths for Windows and macOS
+    //       use the actual file paths supplied by the caller
+    verify_hash(
+        spend_fs,
+        &mut sink,
+        SAPLING_SPEND_HASH,
+        SAPLING_SPEND_BYTES,
+        SAPLING_SPEND_NAME,
```

Comment:
This function is compiled unconditionally, but `SAPLING_SPEND_NAME`, `SAPLING_OUTPUT_NAME`, and `SPROUT_NAME` are conditional on `any(feature = "local-prover", feature = "download-params")`.

Reply:
They need to be unconditional now, because they are used in error messages.

Fixed in 40d9115cc51bdfe4f55c986c30216ce65e8410ea

Reply:
I was confused because compiling `librustzcash` with `--no-default-features` gives different results to `zcash_proofs` by itself, due to inter-crate dependencies. So I submitted a fix that didn't actually work.

I have tested this fix with all the features (and no features) on `zcash_proofs` by itself.